### PR TITLE
Add support for RVV f32 Rdsum microkernels, test and config.

### PR DESCRIPTION
**Pull request by KaustubhIMG**

### CODE REVIEW

**Comment by KaustubhIMG** [comment]
@fbarchard @allenkelly @bhbruce @keeranroth, If you have time, please help review this PR.

File: src/f32-rdsum/gen/f32-rdsum-7p7x-minmax-rvv-u4v.c [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    const float* i1 = (const float*) ((uintptr_t) i0 + input_stride);
+    const float* i2 = (const float*) ((uintptr_t) i1 + input_stride);
+    const float* i3 = (const float*) ((uintptr_t) i2 + input_stride);
+    const float* i4 = (const float*) ((uintptr_t) i3 + input_stride);
+    const float* i5 = (const float*) ((uintptr_t) i4 + input_stride);
+    const float* i6 = (const float*) ((uintptr_t) i5 + input_stride);
+    vfloat32m4_t acc_f32v = __riscv_vfmv_v_f_f32m4(0.f, n);
+
+    for (int r = rows; r > 0; r -= 7) {
+      if XNN_UNPREDICTABLE(r < 2) {
```
Comment (keeranroth) [inline src/f32-rdsum/gen/f32-rdsum-7p7x-minmax-rvv-u4v.c:49]: I still don't like these unpredictable annotations in here. This is very predictable, as we know when we are going to have r < 7. Can this be changed in the code that generates these files?

File: test/f16-f32acc-rdsum.cc [hunk 2]
```diff
@@ -19,58 +19,64 @@
 #if XNN_ENABLE_ARM_FP16_VECTOR && (XNN_ARCH_ARM || XNN_ARCH_ARM64)
   TEST(F16_F32ACC_RDSUM_7P7X__NEONFP16ARITH_C16, channels_eq_16_2pass_fulltile) {
     TEST_REQUIRES_ARM_NEON_FP16_ARITH;
+    const size_t channel_tile = 16;
```
Comment (keeranroth) [inline test/f16-f32acc-rdsum.cc:22]: nit: I think that this is C++, so you can use the `constexpr` keyword. The compiler can tell that this is a compile time constant, so the codegen is not going to be any different, but I like to be more verbose, and show that this is intended to be a compile-time constant

File: tools/generate-rdsum-test.py [hunk 3]
```diff
@@ -27,154 +27,215 @@
...
-  return requantization_type, primary_tile, incremental_tile, channel_tile, arch, isa
+  return requantization_type, primary_tile, incremental_tile, channel_tile, vector_tile, arch, isa
 
 
 RDSUM_TEST_TEMPLATE = """\
-TEST(${TEST_NAME}, channels_eq_${CHANNEL_TILE}_2pass_fulltile) {
+TEST(${TEST_NAME}, channels_eq_${CHANNEL_TILE}${CHANNEL_SUFFIX}_2pass_fulltile) {
   $if ISA_CHECK:
     ${ISA_CHECK};
+  const size_t channel_tile = ${CHANNEL_SCALED_TILE};
```
Comment (keeranroth) [inline tools/generate-rdsum-test.py:50]: nit: I think that this can be `constexpr`, as the code generated is C++?

**Review by keeranroth — APPROVED** [review]
Looks good to me. I've only got some stylistic comments. Feel free to address them, or not.

File: bench/f32-rdsum.cc [hunk 4]
```diff
@@ -55,6 +55,36 @@ BENCHMARK_CAPTURE(f32_rdsum, scalar_c4,
...
 
 
+#if XNN_ENABLE_RISCV_VECTOR && XNN_ARCH_RISCV
+  BENCHMARK_CAPTURE(f32_rdsum, rvv_u1v,
+                    xnn_f32_rdsum_ukernel_7p7x__rvv_u1v,
+                    xnn_init_f32_scaleminmax_scalar_params,
+                    benchmark::utils::CheckRVV)
+    ->Apply(BenchmarkRDSUM)
+    ->UseRealTime();
+#endif  // XNN_ENABLE_RISCV_VECTOR && XNN_ARCH_RISCV
```
[the remark this answers is not in the document]
Comment (KaustubhIMG) [reply bench/f32-rdsum.cc:65]: its generated by scripts/generate-tests-and-benchmarks.sh

File: src/f32-rdsum/gen/f32-rdsum-7p7x-minmax-rvv-u4v.c (hunk 1 above)
Comment (KaustubhIMG) [reply src/f32-rdsum/gen/f32-rdsum-7p7x-minmax-rvv-u4v.c:49]: Yes. I will remove the XNN_UNPREDICTABLE.

**Pull request merged**